gnoi/os/manager.go

```diff
@@ -22,6 +19,7 @@ import (
 // Manager for storing data on OS's.
 type Manager struct {
 	osMap          map[string]bool
+	failMsgs       map[string]string
```

Comment:
also:
activationFailMessage string

Reply:
then below:

```
// SetRunning sets the running OS to the version specified.
func (m *Manager) SetRunning(version string) error {
        if m.activationFailMessage = s.manager.failMsgs[request.Version]; m.activationFailMessage != "" {
		return nil
        }
	if _, ok := m.osMap[version]; ok {
		m.runningVersion = version
		return nil
	}
	return fmt.Errorf("NON_EXISTENT_VERSION")
}
```

Reply:
Also you can add the failMessage as a new parameter in the Install function.

Reply:
Added!